mlir/lib/Target/LLVMIR/DebugTranslation.h

```diff
@@ -102,6 +120,15 @@ class DebugTranslation {
   /// metadata.
   DenseMap<Attribute, llvm::DINode *> attrToNode;
 
+  /// A mapping from DistinctAttr ID of DIRecursiveTypeAttr to the translated
+  /// DIType.
+  llvm::MapVector<DistinctAttr, llvm::DIType *> recursiveTypeMap;
+
+  /// A mapping between DistinctAttr ID and the translated LLVM metadata node.
```

Comment:
```suggestion
  /// A mapping between recursive ID and the translated LLVM metadata node.
```
ultra nit:

Reply:
oh this one actually is the distinct attr itself of nodes that are distinct (this way we don't accidentally create two llvm nodes for the same one because now the same llvm node may appear both inside & outside a loop as two different mlir attrs).

Reply:
Too many IDs :). Ignore my comment then! Or use distinct ID or so...